### DISCUSSION TITLE
Why the provided example gives double  free or corruption (fasttop)

### DISCUSSION THREAD

**Discussion by qiuwei** [nlohmann/json#3004] [Q&A] [answered]
Following the example provided in README, I wrote a simple piece of code:

```cpp
#include <iostream>
#include "json.hpp"
#include <fstream>
#include <iomanip>


int load_config()
{
	//load a conifg from the json file
	using json = nlohmann::json;
	std::ifstream i("ama.json");
	json j;
	i >> j;
	std::ofstream o("pretty.json");
	o << std::setw(4) << j << std::endl;
}

int main() {
	load_config();

}
```

However,  running  it results in the following error:
```
double free or corruption (fasttop)
[1]    2955082 abort (core dumped)  ./build/linux/x86_64/release/test_json
```

Environment:
Ubuntu 20.04
g++ (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

**Comment by nlohmann** [comment, marked as answer]
Apart of `load_config()` not returning a value, that code should be fine. We run all tests with Valgrind and the Google sanitizers, and there are no double frees or any other memory-related issues. Can you please try to get a stack trace or run the code in the debugger?

**Reply by nlohmann** [reply]
@qiuwei Do you have further information on this issue?
